ngraph/core/src/op/reduce_sum.cpp

```diff
@@ -57,10 +58,13 @@ namespace reduce_sum
         bool rc = true;
         switch (arg->get_element_type())
         {
+            NGRAPH_TYPE_CASE(evaluate_reduce_sum, i8, arg, out, axes, keep_dims);
             NGRAPH_TYPE_CASE(evaluate_reduce_sum, i32, arg, out, axes, keep_dims);
             NGRAPH_TYPE_CASE(evaluate_reduce_sum, i64, arg, out, axes, keep_dims);
+            NGRAPH_TYPE_CASE(evaluate_reduce_sum, u8, arg, out, axes, keep_dims);
             NGRAPH_TYPE_CASE(evaluate_reduce_sum, u32, arg, out, axes, keep_dims);
             NGRAPH_TYPE_CASE(evaluate_reduce_sum, u64, arg, out, axes, keep_dims);
+            NGRAPH_TYPE_CASE(evaluate_reduce_sum, bf16, arg, out, axes, keep_dims);
```

Comment:
Why are we adding new evaluate types while they are not needed for constant folding?

Reply:
Agree with Ilya, do we have real cases why we need to support such data types?

Reply:
I have discussed this with @ilyachur and it is not needed. The confusion came because there were backend tests with types i8 and I also saw that CPU plugin supports i8, u8, bf16. Anyway, I have reverted the changes.